tasks/contactdefinitive/app/src/main/java/com/example/contactdefinitive/MainActivity.kt

```diff
@@ -0,0 +1,175 @@
+
+            spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>,
+                    view: View,
+                    position: Int,
+                    id: Long
+                ) {
+                    selectedName1 = parent.getItemAtPosition(position).toString()
+                    coinType1 = selectedName1
+                    if (selectedName1 != "Selecionar") {
+                        Toast.makeText(
+                            this@MainActivity,
+                            selectedName1 + " " + getText(R.string.selected_value_toast),
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    } else {
+                        Toast.makeText(this@MainActivity, R.string.first_toast, Toast.LENGTH_LONG)
+                            .show()
+                    }
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+                    Toast.makeText(
+                        this@MainActivity,
+                        R.string.anything_selected_toast,
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            }
+
+            val items2 = resources.getStringArray(R.array.coins_array2)
+
+            val adapter2 = ArrayAdapter(
+                this@MainActivity,
+                android.R.layout.simple_spinner_dropdown_item,
+                items2
+            )
+            adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+            spin2.adapter = adapter2
+
+            spin2.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>,
+                    view: View,
+                    position: Int,
+                    id: Long
+                ) {
+                    selectedName2 = parent.getItemAtPosition(position).toString()
+                    coinType2 = selectedName2
+                    if (selectedName2 != "Selecionar") {
+                        Toast.makeText(
+                            this@MainActivity,
+                            selectedName2 + " " + getText(R.string.selected_value_toast),
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    } else {
+                        Toast.makeText(this@MainActivity, R.string.first_toast, Toast.LENGTH_LONG)
+                            .show()
+                    }
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+                    Toast.makeText(
+                        this@MainActivity,
+                        R.string.anything_selected_toast,
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            }
```

Comment:
you can move this to setupSpinners() and call it from setupListeners.
Try to simplify the expressions to avoid unnecessary code.